docker/templates/hardening_manifest.yaml.erb

```diff
@@ -44,16 +44,6 @@ resources:
   validation:
     type: sha512
     value: <INSERT SHA512 VALUE FROM https://artifacts.elastic.co/downloads/logstash/logstash-<%= elastic_version %>-linux-x86_64.tar.gz.sha512>
-- filename: go1.17.8.linux-amd64.tar.gz
-  url: https://dl.google.com/go/go1.17.8.linux-amd64.tar.gz
-  validation:
-    type: sha256
-    value: 980e65a863377e69fd9b67df9d8395fd8e93858e7a24c9f55803421e453f4f99
-- filename: v2.3.0.tar.gz
```

Comment:
We build env2yaml with `go build` now, while the old way is `go build -mod vendor` taking the source from vendor.
v2.3.0.tar.gz is no longer need